### PR TITLE
maint: upgrade from go1.x to provided.al2

### LINKED ISSUE #63: AWS is deprecating the `go1.x` runtime
**Is your feature request related to a problem? Please describe.**

The `rds-logs` Terraform module uses the `go1.x` runtime. AWS is ending support for this runtime on December 31st, 2023 and recommends migrating to the `provided.al2` runtime instead.

**Describe the solution you'd like**

Upgrade the runtime here:

https://github.com/honeycombio/terraform-aws-integrations/blob/4d8a20f14340c99622be71709b2ecf472dd12672/modules/rds-logs/main.tf#L39

https://github.com/honeycombio/terraform-aws-integrations/blob/4d8a20f14340c99622be71709b2ecf472dd12672/modules/s3-logfile/main.tf#L45

(Or if there are major differences, provide a config flag?)

**Describe alternatives you've considered**

N/A.

**Additional context**

Here is the full email that AWS sent us:

> Hello,
>
> We are contacting you as we have identified that your AWS Account currently has one or more Lambda functions using the Go 1.x runtime.
> 
> We are ending support for the Go 1.x runtime in AWS Lambda, in line with Amazon Linux AMI end of maintenance support on December 31, 2023 [1]. Lambda continues to support the Go programming language, using the provided.al2 runtime. Using the provided.al2 runtime offers several benefits over the go1.x runtime, including support for AWS Graviton2 processors and a streamlined implementation with a smaller deployment package and faster function invoke path. For more information, see our blog post [2].
>
> As described in the Lambda runtime support policy [3], end of support for language runtimes in Lambda happens in two stages. Starting December 31, 2023, Lambda will no longer apply security patches and other updates to the Go 1.x runtime used by Lambda functions, and functions using Go 1.x will no longer be eligible for technical support. In addition, you will no longer be able to create new Lambda functions using the Go 1.x runtime. Starting January 30, 2024, you will no longer be able to update existing functions using the Go 1.x runtime.
> 
> We recommend that you upgrade your existing Go 1.x functions to the provided.al2 runtime before December 31, 2023.
> 
> End of support does not impact function execution. Your functions will continue to run. However, they will be running on an unsupported runtime which is no longer maintained or patched by the AWS Lambda team.
>
> This notification is generated for functions using the Go 1.x runtime for the $LATEST function version. A list of your impacted Lambda functions can be found in the 'Affected resources' tab of your Personal Health Dashboard. The following command shows how to use the AWS CLI [4] to list all functions in a specific region using Go 1.x, including published function versions. To find all such functions in your account, repeat this command for each region:
>
>     aws lambda list-functions --function-version ALL --region us-east-1 --output text --query "Functions[?Runtime=='go1.x'].FunctionArn"
> 
> There is a difference in Lambda billing between the go1.x runtime and the provided.al2 runtime. With the go1.x runtime, Lambda does not bill for time spent during function initialization (cold start), whereas with the provided.al2 runtime Lambda includes function initialization time in the billed function duration. Since Go functions typically initialize very quickly, and since Lambda reduces the number of initializations by re-using function execution environments for multiple function invokes, in practice the difference in your Lambda bill should be very small.
> 
> If you have any concerns or require further assistance, please contact AWS Support [5].
> 
> [1] https://aws.amazon.com/blogs/aws/update-on-amazon-linux-ami-end-of-life/
> [2] https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-from-the-go1-x-runtime-to-the-custom-runtime-on-amazon-linux-2/
> [3] https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html
> [4] https://aws.amazon.com/cli/
> [5] https://aws.amazon.com/support
```

### PR DESCRIPTION
**Pull request by brookesargent**
## Which problem is this PR solving?

- Closes #63 

## Short description of the changes

Uses the new individual zip files generated from `agentless-aws-integrations` v4.0.0 and sets the runtime to `provided.al2` due to `go1.x` being deprecated at the end of this year. Also adds an argument for architecture so `x86_64` or `arm64` can be used but `arm64` is the default.

Once this goes through an initial round of code review, will test our own usage of the module with this new version prior to merging.

### CODE REVIEW

File: modules/rds-logs/variables.tf [hunk 1]
```diff
@@ -133,6 +133,17 @@ variable "http_buffering_interval" {
   description = "Kinesis Firehose http buffer interval, in seconds."
 }
 
+variable "lambda_function_architecture" {
+  type        = string
+  default     = "arm64"
```
Comment (jharley) [inline modules/rds-logs/variables.tf:138]: If we make the default amd64 then this is probably a smoother upgrade path and could even be only a minor release of the module? 🤔

**Review by jharley — APPROVED** [review]
This is lookin' good to me! Left a suggestion about leaving the architecture as amd64 by default.

File: modules/rds-logs/variables.tf (hunk 1 above)
Comment (brookesargent) [reply modules/rds-logs/variables.tf:138]: I'm open to that! One of my main motivators to do arm was for the cost/performance benefits but tbh I don't know how significant those will be so if we think amd is better, I'll change it!

Comment (jharley) [reply modules/rds-logs/variables.tf]: I think they're great benefits, but, in the name of not surprising current consumers of the module I thought amd64 made sense as the default (as it's current behaviour)? Happy to push ARM64 here though ✨

Comment (brookesargent) [reply modules/rds-logs/variables.tf]: Agree, I think it's worth the tradeoff for the smoother upgrade path. I pushed the change!

**Pull request merged**